### PR TITLE
Test check title and count hot seller products

### PR DESCRIPTION
**Pull request by RaduMoldovan30**
Tried different methods and commented code, only for learning purposes.

### CODE REVIEW

File: tests/homePage.spec.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+test('count products on the home page', async ({ page }) => {
+    await page.goto('https://magento.softwaretestingboard.com/')
+
+    // Count the number of elements with the class 'product-item'
+    const numberOfProducts = await page.locator('.product-item').evaluateAll(products => products.length);
+    console.log('Number of products on the home page is:', numberOfProducts);
+
+    // Expect the number of products to be greater than 0
+    expect(numberOfProducts).toBeGreaterThanOrEqual(6)
```
Comment (adrianmaciuc) [inline tests/homePage.spec.js:34]: The requirements says to be 6. Not greater than.

**Review by RaduMoldovan30 — COMMENTED** [review]
Sorry, I commit another test....

**Review by simonleventeattila — APPROVED** [review]

File: tests/example.spec.js [hunk 2]
```diff
@@ -17,3 +17,4 @@ test('get started link', async ({ page }) => {
   // Expects page to have a heading with the name of Installation.
   await expect(page.getByRole('heading', { name: 'Installation' })).toBeVisible();
 });
+
```
Comment (AdrianCiuciui) [inline tests/example.spec.js:20]: why do you have a plus symbol here?

File: tests/homePage.spec.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// @ts-check
+const { test, expect } = require('@playwright/test');
+const { chromium } = require('playwright');
+
+
+test('Check if the user is on the Home Page', async ({ page }) => {
+
+    await page.goto('https://magento.softwaretestingboard.com/')
+    //await expect(page).toHaveURL('https://magento.softwaretestingboard.com/') --> 1st choice
```
Comment (AdrianCiuciui) [inline tests/homePage.spec.js:9]: why is there so much commented code?

File: tests/homePage.spec.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    //console.log('Home Page is visible: ', logoImg)
+
+    // Get the text content of the entire page
+    const pageTextContent = await page.textContent('html');
+
+    // Check if the page contains the specified text
+    const containHomePage = pageTextContent.includes('Home Page');
+    console.log('Home Page is visible: ', containHomePage)
+
+    //I did all this only for learning, but the perfect way is to check Url and Title, in this way the user confirm that is on the Home Page
```
Comment (AdrianCiuciui) [inline tests/homePage.spec.js:23]: I don't think this comment is needed

Comment (RaduMoldovan30) [reply tests/homePage.spec.js:23]: I added it only for learning puroses, I will remove it

File: tests/homePage.spec.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    await page.goto('https://magento.softwaretestingboard.com/')
+  
+    const numberOfProducts = await page.$$('.product-item');
+    const randomProduct = numberOfProducts[Math.floor(Math.random() * numberOfProducts.length)];
+    const randomTitleBeforeClick = await randomProduct.evaluate(product => product.querySelector('.product-item-link').textContent.trim());
+    const titleName = randomTitleBeforeClick;
+    await randomProduct.click();
+
+  
+    const randomTitleAfterRedirect = await page.evaluate(() => document.querySelector('span[data-ui-id="page-title-wrapper"][itemprop="name"]').textContent.trim());
```
Comment (AdrianCiuciui) [inline tests/homePage.spec.js:31]: The issue I mentioned in comment on line 26 also occurs on this line, for the document.

File: tests/homePage.spec.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    console.log('Number of products on the home page is:', numberOfProducts);
+    expect(numberOfProducts).toBe(6)
+});
+
+test('Select random products from home page', async ({ page }) => {
+    await page.goto('https://magento.softwaretestingboard.com/')
+  
+    const numberOfProducts = await page.$$('.product-item');
+    const randomProduct = numberOfProducts[Math.floor(Math.random() * numberOfProducts.length)];
+    const randomTitleBeforeClick = await randomProduct.evaluate(product => product.querySelector('.product-item-link').textContent.trim());
```
[the remark this answers is not in the document]
Comment (AdrianCiuciui) [reply tests/homePage.spec.js:26]: If you run it multiple times it should fail on your side too

**Review by AdrianCiuciui — CHANGES_REQUESTED** [review]
Hope it helps

File: tests/homePage.spec.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    await expect(page).toHaveTitle("Home Page"); 
+});
+
+
+test('count products on the home page', async ({ page }) => {
+
+    await page.goto('https://magento.softwaretestingboard.com/');
+    const numberOfProducts = await page.locator('.product-item').evaluateAll(products => products.length);
+    console.log('Number of products on the home page is:', numberOfProducts);
+    expect(numberOfProducts).toBe(6)
```
[the remark this answers is not in the document]
Comment (adrianmaciuc) [reply tests/homePage.spec.js:18]: Do not add ; at the end, this is not Java :) . that sign is optional in javascript.

File: tests/homePage.spec.js (hunk 5 above)
Comment (RaduMoldovan30) [reply tests/homePage.spec.js:31]: I know these problems guys, but I stayed too many hours on this task, and I did not know how to resolve the issues on lines with .evaluate. I need some guidance from who knows how to fix it. Thanks

**Comment by RaduMoldovan30** [comment]
The pull request has been closed because the branch was deleted. A new branch will be created

**Pull request closed without merging**